examples/watrli_rpl_udp/main.c

```diff
@@ -0,0 +1,264 @@
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       watr.li rpl udp application
+ *
+ * @author
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <unistd.h>
+#include "udp.h"
+#include "rpl.h"
+
+#define WATR_LI_CHANNEL         (26)     /**< The used channel */
+#define WATR_LI_PAN             (0x03e9) /**< The used PAN ID */
+#define WATR_LI_IFACE           (0)      /**< The used Trasmssion device */
+#define WATR_LI_UDP_PORT        (12345)  /**< The UDP port to listen */
+#define WATR_LI_UDP_BUFFER_SIZE (1024)   /**< The buffer size for receiving UDPs */
+
+/** The UDP server thread stack */
+char udp_server_stack_buffer[KERNEL_CONF_STACKSIZE_MAIN];
+/** The node IPv6 address */
+ipv6_addr_t myaddr;
+
+/**
+* @brief the sample UDP server that expects receiving strings
+* @param[in] arg unused parameter pointer
+*/
+static void *watr_li_udp_server(void *arg)
+{
+    (void) arg;
+
+    sockaddr6_t sa;
+    char buffer_main[WATR_LI_UDP_BUFFER_SIZE];
+    uint32_t fromlen;
+    int sock = socket_base_socket(PF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+
+    memset(&sa, 0, sizeof(sa));
+
+    sa.sin6_family = AF_INET;
+    sa.sin6_port = HTONS(WATR_LI_UDP_PORT);
+
+    fromlen = sizeof(sa);
+
+    if (-1 == socket_base_bind(sock, &sa, sizeof(sa))) {
+        puts("[watr_li_udp_server] Error bind failed!");
+        socket_base_close(sock);
+        return NULL;
+    }
+
+    while (1) {
+        int32_t recsize = socket_base_recvfrom(sock, (void *)buffer_main,
+                                               WATR_LI_UDP_BUFFER_SIZE, 0,
+                                               &sa, &fromlen);
+
+        if (recsize < 0) {
+            puts("[watr_li_udp_server] ERROR: recsize < 0!");
+        } else {
+            /* if we received a string print it */
+            if (buffer_main[recsize-1] == '\0' ) {
+                printf("UDP packet received, payload:\n%s\n", buffer_main);
+            } else {
+                /* print the buffer bytes in hex */
+                printf("UDP packet received, payload (%d bytes):\n", (int)recsize);
+                for(int i = 0; i < recsize; ++i) {
+
+                    if ( (i%8) == 0 ) {
+                        /* newline after 8 bytes */
+                        puts("");
+                    }
+
+                    printf("%02x ", buffer_main[i]);
+                }
+                puts("");
+            }
+        }
+
+    }
+
+    socket_base_close(sock);
+
+    return NULL;
+}
+
+/**
+* @brief sends a packet to the DODAG ID (should be the root node IPv6 address)
+* @param[in] payload pointer to the payload to be sent
+* @param[in] size number of bytes of the payload
+*/
+static void watr_li_udp_send(char* payload, size_t payload_size)
+{
+    int sock;
+    sockaddr6_t sa;
+    int bytes_sent;
+    sock = socket_base_socket(PF_INET6, SOCK_DGRAM, IPPROTO_UDP);
```

Comment:
The main has changed a bit in the plant_node master anyway.. ;) I'd create the socket in watr_li_setup_node() and close it at the end of my new main. Thanks for the quick response!